gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts

```diff
@@ -0,0 +1,180 @@
+
+import { logger, BN } from '@chainlink/gauntlet-core/dist/utils'
+import { PublicKey, SYSVAR_RENT_PUBKEY, Keypair } from '@solana/web3.js'
+import { CONTRACT_LIST, getContract } from '@chainlink/gauntlet-solana-contracts'
+import { Idl, Program } from '@project-serum/anchor'
+
+export const wrapCommand = (command) => {
+  return class Multisig extends SolanaCommand {
+    command: SolanaCommand
+    program: Program<Idl>
+    proposal: PublicKey
+    multisigAddress: PublicKey
+    multisigSigner: PublicKey
+    owners: [PublicKey]
+    threshold: number
+
+    static id = `${command.id}`
+
+    constructor(flags, args) {
+      super(flags, args)
+      logger.info(`Running ${command.id} command using Serum Multisig`)
+
+      this.command = new command(flags, args)
+      this.command.invokeMiddlewares(this.command, this.command.middlewares)
+    }
+
+    execute = async () => {
+      this.require(process.env.MULTISIG_ADDRESS != null, 'Please set MULTISIG_ADDRESS env var')
+      this.multisigAddress = new PublicKey(process.env.MULTISIG_ADDRESS || '')
+      const multisig = getContract(CONTRACT_LIST.MULTISIG, '')
+      this.program = this.loadProgram(multisig.idl, multisig.programId.toString())
+      logger.info(`Multisig Address: ${process.env.MULTISIG_ADDRESS}`)
+      const [multisigSigner] = await PublicKey.findProgramAddress(
+        [this.multisigAddress.toBuffer()],
+        this.program.programId,
+      )
+      this.multisigSigner = multisigSigner
+      logger.info(`Multisig Signer: ${multisigSigner.toString()}`)
+      const multisigState = await this.program.account.multisig.fetch(process.env.MULTISIG_ADDRESS)
+      logger.debug('Multisig state:')
+      logger.debug(JSON.stringify(multisigState, null, 4))
+      this.threshold = multisigState.threshold
+      this.owners = multisigState.owners
+
+      const isCreation = !this.flags.proposal
+      if (isCreation) {
+        const rawTxs = await this.command.makeRawTransaction(this.multisigSigner)
+        return this.wrapAction(this.createProposal(rawTxs[0]))
+      }
+
+      const proposalState = await this.program.account.transaction.fetch(this.flags.proposal)
+      logger.debug('Proposal state:')
+      logger.debug(JSON.stringify(proposalState, null, 4))
+
+      const isAlreadyExecuted = proposalState.didExecute
+      if (isAlreadyExecuted == true) {
+        logger.info(`Proposal is already executed`)
+        return {} as Result<TransactionResponse>
+      }
+
+      const remainingApprovalsNeeded = this.remainingApprovalsNeeded(proposalState, this.threshold)
+      const isReadyForExecution = remainingApprovalsNeeded > 0 ? false : true
+      if (isReadyForExecution) {
+        return this.wrapAction(this.executeProposal(proposalState))
+      }
+
+      return this.wrapAction(this.approveProposal())
+    }
+
+    wrapAction = async (action) => {
+      try {
+        const tx = await action
```

Comment:
`action` seems like it's being executed already. Try to wrap the function first, then add the args you want for this new fn  
```suggestion
    type ProposalAction = (rawTx: RawTransaction) => Promise<string>
    ...
    wrapAction = (proposalAction: ProposalAction) => (rawTx: RawTransaction) {
      try {
        const tx = await proposalAction(rawTx)
```
You would consume it like:
```js
this.wrapAction(this.createProposal)(rawTxs[0])
```